referenceProvider.js

```diff
@@ -0,0 +1,128 @@
+const vscode = require('vscode');
+const parser = require('./parser');
+const search = require('./search');
+
+exports.ReferenceProvider = class {
+    provideDefinition(document, position) {
+        const fullText = document.getText();
+        const currentFilePath = document.fileName;
+        const range = document.getWordRangeAtPosition(position);
+
+        let childWord = "";
+        let moduleList = {};
+
+        if(range) {
+            const textAtCarrot = document.getText(range);
+            const fullTextWithoutComments = parser.removeComments(fullText);
+            const requireOrDefineStatement = parser.getRequireOrDefineStatement(fullTextWithoutComments);
+
+            if(requireOrDefineStatement) {
+                moduleList = parser.getModulesWithPathFromRequireOrDefine(requireOrDefineStatement);
+            }
+
+            const modulePath = textAtCarrot in moduleList ? moduleList[textAtCarrot] : null;
+
+            //We matched a module (word is a module)
+            if(modulePath) {
+                let searchFor = '';
+                let stopSearchingFurther;
+
+                if(childWord == "") {//Not a parent - search for the module name (word)
+                    searchFor = textAtCarrot;
+                    stopSearchingFurther = true;
+                } else { //It is a parent, search for the child which is a property of the module
+                    searchFor = childWord;
+                    stopSearchingFurther = false;
+                }
+                childWord = "";
+
+                return search.forModule(currentFilePath, modulePath, searchFor, stopSearchingFurther);
+            } else { //word is not a module
+                return new Promise(resolve => {
+                    let continueFrom;
+
+                    let dotPosition = document.offsetAt(new vscode.Position(range._start._line, range._start._character-1));
+
+                    //Do backwards search for a dot
+                    dotPosition = fullTextWithoutComments.lastIndexOf('.', dotPosition);
```

Comment:
The old method also returns false if there is a character other than a space or tab or new line. Are you sure this matches the same functionality? I doubt it.